addons/product/models/product_product.py

```diff
@@ -231,7 +228,7 @@ def _check_duplicated_product_barcodes(self, barcodes_within_company, company_id
 
     def _check_duplicated_packaging_barcodes(self, barcodes_within_company, company_id):
```

Comment:
Since we're checking it in the `product.uom` model, do we still need to check it here?

Reply:
Yes, the check here is enforced when the product barcode is set/changed. The other check is enforced when a packaging barcode is set/changed. Therefore, I think both are needed.

Reply:
Should have read the comment better, that's correct. :+1: